adr/0007-choosing-an-open-source-component.md

```diff
@@ -0,0 +1,57 @@
+# Choosing an open source component
+
+Date: 2018-01-09
+
+## Status
+
+Proposed.
+
+## Context 
+
+Using poor quality or unmaintained third-party software can be at the least inconvenient, and at the most downright dangerous: for example, it can fail to take advantage of the benefits from the latest technology, and security holes can go unpatched.
+
+Using third-party software that is difficult to maintain integrated in our environment can make updating it difficult, time-consuming or risky.
+
+Using software with an inappropriate license may present a legal risk, or may risk others choosing not to use our software because of the burden the license subsequently places on them.  
+
+We want to minimise exposure to these risks when choosing an open source component to use in the eLife code base.   
+            
+## Decision
+When choosing an open source component, care should be taken to try to choose something that is:
+
+- licensed appropriately
+- reliable
+- well maintained
+- sustainable within our environment
+
+### Licensing
+
+[should we specify this in terms of an explicit list of acceptable licenses, or in terms of what we need a licence to allow us to do?]
```

Comment:
considering we release everything with the MIT license, there is a compatibility issue (with GPL?)

Reply:
an explanation on what it means to be 'compatible' with GPL licensed code: https://www.gnu.org/licenses/gpl-faq.en.html#WhatIsCompatible

and a list of compatible licences:
https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses

there is a lot of argument and untested legal theorizing on edge cases in most software licences. The only perfectly clear cut legal standing, in my humble opinion, is "public domain". Even the legality of closed sourced proprietary code becomes fuzzy if it's not 100% written internally in the same country in the same decade.

also, there are different types of code usage, like linking vs bundling vs compiling. is a compiled program with gpl licensed code required to also be licenced by the GPL? (yes). what if it's unmodified and uncompiled, or compiled to an intermediate format for an interpreter (javascript, python)? what if it's called from a script? is a script a program? all sorts of fun and games here.

Ian settled on the MIT licence for eLife, and while I don't agree with that decision ideologically, it is convenient for others for all sorts of reasons. Convenience and fewer barriers and very important to adoption if adoption is one of your goals.

I would be curious to find the licences of all of the dependencies our programs use and see which large groups emerge. I'm less worried about proprietary software sneaking in than I am about those who use a licence like MIT or BSD and then 'tweak' it, adding their own custom clauses.

+1 for a licence audit. I know observer and metrics and possibly lax are still GPLd.